From ca4942e89cf462e5f8c36ca9b8b689879083af6b Mon Sep 17 00:00:00 2001
From: David Henningsson <david.henningsson@canonical.com>
Date: Tue, 16 Apr 2013 06:40:40 +0200
Subject: [PATCH] alsa-mixer: Switch to Headphone when Headphone mic jack is
 plugged in

When a "Headphone Mic" jack becomes available, we do not know if
a headphone or a mic has been plugged in. Therefore, setting both
paths to "unknown" is, in theory, the correct thing to do.
However, in practice, people are more likely to plug in a headphone
rather than a mic. Therefore, allow autoswitch to the headphone port
when the jack is plugged in.

A more advanced implementation would consider what was plugged in last
time depending on what port was selected on the input side at that
time, and set availability accordingly. However, such an implementation
will have to wait (probably at least until we have our fancy routing
system implementation).

Buglink: https://bugs.launchpad.net/bugs/1169143
Signed-off-by: David Henningsson <david.henningsson@canonical.com>
---
 .../alsa/mixer/paths/analog-output-headphones.conf |    1 -
 1 file changed, 1 deletion(-)

Index: pulseaudio-3.0/src/modules/alsa/mixer/paths/analog-output-headphones.conf
===================================================================
--- pulseaudio-3.0.orig/src/modules/alsa/mixer/paths/analog-output-headphones.conf	2013-07-24 09:31:32.174044445 +0200
+++ pulseaudio-3.0/src/modules/alsa/mixer/paths/analog-output-headphones.conf	2013-07-24 09:40:19.106035129 +0200
@@ -34,7 +34,6 @@
 # This jack can be either a headphone *or* a mic. Used on some ASUS netbooks.
 [Jack Headphone Mic]
 required-any = any
-state.plugged = unknown
 
 [Element Hardware Master]
 switch = mute
